theseus/utils/examples/motion_planning/motion_planner.py

```diff
@@ -337,36 +339,58 @@ def get_variable_values_from_straight_line(
     ) -> Dict[str, torch.Tensor]:
         # Returns a dictionary of variable names to values that represent a straight
         # line trajectory from start to goal.
-        start_goal_dist = goal - start
+        # For SE2 variables, the start's angle is used for the full trajectory
+        start_goal_dist = goal[:, :2] - start[:, :2]
         avg_vel = start_goal_dist / self.objective.total_time
         unit_trajectory_len = start_goal_dist / (self.objective.trajectory_len - 1)
         input_dict: Dict[str, torch.Tensor] = {}
         for i in range(self.objective.trajectory_len):
-            input_dict[f"pose_{i}"] = start + unit_trajectory_len * i
-            input_dict[f"vel_{i}"] = avg_vel
+            if self.objective.pose_type == th.SE2:
+                cur_pos = start[:, :2] + unit_trajectory_len * i
+                input_dict[f"pose_{i}"] = torch.cat([cur_pos, start[:, 2:]], dim=1)
+                input_dict[f"vel_{i}"] = torch.cat(
+                    [avg_vel, torch.zeros(avg_vel.shape[0], 1)], dim=1
+                )
+            else:
+                input_dict[f"pose_{i}"] = start + unit_trajectory_len * i
+                input_dict[f"vel_{i}"] = avg_vel
         return input_dict
 
-    def get_random_variable_values(
-        self, start: torch.Tensor
+    def get_randn_trajectory_like(
+        self,
+        start: torch.Tensor,
     ) -> Dict[str, torch.Tensor]:
         # Returns a dictionary of variable names with random initial poses.
+        # The batch size, device and dtype are obtained from given start
+        pose_numel = 2 if self.objective.pose_type == th.Point2 else 4
+        vel_numel = 2 if self.objective.pose_type == th.Point2 else 3
         input_dict: Dict[str, torch.Tensor] = {}
+        assert start.shape[1] == pose_numel
         for i in range(self.objective.trajectory_len):
             input_dict[f"pose_{i}"] = torch.randn_like(start)
-            input_dict[f"vel_{i}"] = torch.randn_like(start)
+            input_dict[f"vel_{i}"] = torch.randn_like(start[:, :vel_numel])
         return input_dict
 
     def get_variable_values_from_trajectory(
         self, trajectory: torch.Tensor
     ) -> Dict[str, torch.Tensor]:
         # Returns a dictionary of variable names to values, so that values
-        # are assigned with the data from the given trajectory. Trajectory should be a
+        # are assigned with the data from the given trajectory.
+        # For Point2 trajectories, trajectory should be a
         # tensor of shape (batch_size, 4, planner.trajectory_len).
-        assert trajectory.shape[1:] == (4, self.objective.trajectory_len)
+        # For SE2 trajectories, it should be a tensor of shape
+        # tensor of shape (batch_size, 7, planner.trajectory_len).
+        pose_numel = 2 if self.objective.pose_type == th.Point2 else 4
```

Comment:
you could use the Manifold.numel() available in main now?

Reply:
Changed.